### ISSUE TITLE
CloudBuild trigger fails

### ISSUE THREAD

**Issue by tszebeni** [GoogleCloudPlatform/inspec-gcp-cis-benchmark#100]
Dear repo maintainers,
I've run into issues when I wanted to automate cis benchmark for a project on GCP via the option you provide (by using your cloudbuild.yaml).
Here are my steps:
I forked the repo to my account and linked Cloud Source Repositories to it. I created a Cloud Build trigger to use cloudbuild.yaml provided in the repository, I set substitution variables gcp project id and bucket, I provided permission to service account to write into the bucket, and I clicked run. Unfortunately the run cannot be started, the error message does not tell me any specific reason, please advise. Thanks
`
Error message: "Failed to trigger build: cloud.scoping-monitoring/github_tszebeni_inspec-gcp-cis-benchmark:cloudbuild.yaml:"
`
Log entry:
```
{
  "protoPayload": {
    "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
    "status": {
      "code": 5,
      "message": "cloud.scoping-monitoring/github_tszebeni_inspec-gcp-cis-benchmark:cloudbuild.yaml: "
    },
    "authenticationInfo": {
      "principalEmail": "tamas.szebeni@xyz",
      "principalSubject": "user:tamas.szebeni@xyz"
    },
    "requestMetadata": {
      "callerIp": "xyz",
      "callerSuppliedUserAgent": "xyz",
      "requestAttributes": {},
      "destinationAttributes": {}
    },
    "serviceName": "cloudbuild.googleapis.com",
    "methodName": "google.devtools.cloudbuild.v1.CloudBuild.RunBuildTrigger",
    "authorizationInfo": [
      {
        "resource": "projects/scoping-monitoring",
        "permission": "cloudbuild.builds.create",
        "granted": true,
        "resourceAttributes": {}
      }
    ],
    "resourceName": "projects/scoping-monitoring/triggers/fc6ecf13-35c8-46bf-9e32-7ecbc5b79a57",
    "request": {
      "name": "projects/scoping-monitoring/locations/europe-west3/triggers/fc6ecf13-35c8-46bf-9e32-7ecbc5b79a57",
      "projectId": "scoping-monitoring",
      "triggerId": "fc6ecf13-35c8-46bf-9e32-7ecbc5b79a57",
      "source": {
        "branchName": "master",
        "substitutions": {
          "_GCP_PROJECT_ID": "scoping-monitoring",
          "_REPORTS_BUCKET": "cis-benchmark-reports-xyz"
        }
      },
      "@type": "type.googleapis.com/google.devtools.cloudbuild.v1.RunBuildTriggerRequest"
    }
  },
  "insertId": "134ocq5bwp",
  "resource": {
    "type": "build",
    "labels": {
      "build_trigger_id": "fc6ecf13-35c8-46bf-9e32-7ecbc5b79a57",
      "build_id": "",
      "project_id": "scoping-monitoring"
    }
  },
  "timestamp": "2023-05-17T06:55:52.319464578Z",
  "severity": "ERROR",
  "logName": "projects/scoping-monitoring/logs/cloudaudit.googleapis.com%2Factivity",
  "receiveTimestamp": "2023-05-17T06:55:52.712428385Z"
}
```
See the status entry above.

Thank you

**Comment by tszebeni** [comment]
![image](https://github.com/GoogleCloudPlatform/inspec-gcp-cis-benchmark/assets/328690/7108a233-6eed-45e9-8347-d142e222e302)

**Comment by KonradSchieban** [comment]
Hi Tamas, looking at the audit log entry, it looks like you are missing the permission cloudbuild.builds.create . Have you checked that you have the appropriate IAM roles assigned to your account (e.g. one of the pre-defined roles from [here](https://cloud.google.com/build/docs/iam-roles-permissions)).

Your issue does not looks specific to the repository.. Can you trigger **any** Cloud Build pipeline?

Otherwise, there might also be the chance that the connection between Cloud Build and Github is not authorized. Have you followed the steps from [this doc](https://cloud.google.com/build/docs/automating-builds/github/connect-repo-github)?

**Comment by tszebeni** [comment]
Hi @KonradSchieban , thanks for the answer.
It turned out that Cloud Build expects the cloudbuild.yml filename with extension yml (without 'a'), in the repository it is named differently so I had to change that in the trigger config or rename a file in my fork.
I also had to make some customization to fix cloudbuild.yml like reorganizing the compliance and last steps, so if compliance failing, the json report would still uploaded. I'm working on creating a log based metric to provide monitoring dashboard based on the test result now.
